source/common/upstream/health_checker_impl.h

```diff
@@ -152,7 +154,7 @@ class HealthCheckerImplBase : public HealthChecker,
 
   static const std::chrono::milliseconds NO_TRAFFIC_INTERVAL;
 
-  std::list<HostStatusCb> callbacks_;
+  mutable std::list<HostStatusCb> callbacks_;
```

Comment:
Code smell?

Reply:
Yeah, I'm not super thrilled with this either, though this is already done in HostSet for member changed callbacks. One could argue that the constness of the thing does not have to do with receiving callbacks about it changing. I could take a look at reworking all of this but not sure if it's really worth it. LMK.

Reply:
I'm guessing you had to do this because it's now used in some place where we only have a const ref? Maybe just a TODO and call it a day, I agree there isn't a lot we gain right now here. I think there is a fair bit of const confusion in Envoy that would be good to cleanup one day; stats is one other place.

Reply:
Sure can add some TODOs. C++ constness is far from perfect which IMO makes "perfection" in this area almost impossible.